File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] s) {
       if(s==null||s.length==0)
       {
           return "";
       } 

       String pre = s[0];
       for(int i=1;i<s.length;i++)
       {
           while(!s[i].startsWith(pre))
           {
               pre= pre.substring(0,pre.length()-1);
           }
       }
       return pre;
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& str) 
    {
        string ans = "";
        int j=0;
        while(true)
        {
            if(str[0].length()<=j)
            {
                return ans;
            }
            
            char temp = str[0][j];
             
            for(int i=1; i<str.size(); i++)
            {
                if(str[i].length()>j)
                {
                    if(str[i][j]!=temp)
                    {
                        return ans;
                    }
                }
                else
                    return ans;
            }
            ans = ans + temp;
            j++;
        }

        return ans;    
    }
};